--- fluidsynth/src/drivers/fluid_dsound.c.orig	2012-08-16 00:01:13.000000000 -0400
+++ fluidsynth/src/drivers/fluid_dsound.c	2015-08-16 19:57:19.062921700 -0400
@@ -111,10 +111,16 @@
   fluid_dsound_devsel_t devsel;
 
   /* check if the globals are initialized */
+#ifndef FLUIDSYNTH_NOT_A_DLL
   if (FLUID_HINSTANCE == NULL) {
     FLUID_LOG(FLUID_ERR, "FluidSynth hinstance not set, which is needed for DirectSound");
     return NULL;
   }
+#else
+  HINSTANCE hinstance = GetModuleHandle(NULL);
+  fluid_set_hinstance(hinstance);
+  fluid_win32_create_window();
+#endif
 
 /*
   if (fluid_wnd == NULL) {
